WORKSPACE

```diff
@@ -41,8 +41,14 @@ load("//library/maven:rules.bzl", "maven")
 # Load //builder/kotlin
 load("//builder/kotlin:deps.bzl", kotlin_deps = "deps")
 kotlin_deps()
-load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kotlin_repositories", "kt_register_toolchains")
-kotlin_repositories()
+load("@io_bazel_rules_kotlin//kotlin:repositories.bzl", "kotlin_repositories", "kotlinc_version")
+kotlin_repositories(
+    compiler_release = kotlinc_version(
+        release = "1.7.20",
+        sha256 = "5e3c8d0f965410ff12e90d6f8dc5df2fc09fd595a684d514616851ce7e94ae7d"
+    )
+)
```

Comment:
We specify our kotlinc version explicitly now, along with the SHA256 hash of the release.

Reply:
Interesting. Is this an approach favoured by `rules_kotlin` now?

In Rust, originally, we specified our `rustc` version explicitly, and then we changed that behaviour because each release of `rules_rust` has its own default `rustc` version number which doesn't change until we upgrade `rules_rust` again. Having said that, I'm not certain that was the right approach for us. On the one hand, it's good to be explicit about language versions; on the other hand, we (almost!) always just want the latest version, and in the latter case, it's arguably better for there to be a single source of truth for the language version - and that's whatever version is specified in `rules_rust` (or in our case here, `rules_kotlin`).

Reply:
I was mistaken having this here: we specify the version of kotlinc per WORKSPACE. dependencies should only be making the appropriate version of rules_kotlin available.

Reply:
Hm. It makes sense that we can specify the `kotlinc` version per `WORKSPACE`; however, `dependencies` is also a self-contained repo that we are supposed to be able to run `bazel build //...` on and get a deterministic output. And it most certainly does use Kotlin binaries to build the project.

Given, therefore, that the build succeeds without a `kotlinc` version specified, I suppose we can assume that `rules_kotlin` behaves similarly to `rules_rust`, having a default `kotlinc` version that it uses?

Could we verify this?

Reply:
> Given, therefore, that the build succeeds without a kotlinc version specified, I suppose we can assume that rules_kotlin behaves similarly to rules_rust, having a default kotlinc version that it uses?
That's correct. When we call `kotlin_repositories` with no arguments, a default kotlinc version is chosen based on some hardcoded values in rules_kotlin's `//src/main/starlark/core/repositories:versions.bzl`.